egg/core/callbacks.py

```diff
@@ -131,11 +134,23 @@ def __init__(
         checkpoint_path: Union[str, pathlib.Path],
         checkpoint_freq: int = 1,
         prefix: str = "",
+        start_epoch: int = 0,
+        max_chk: int = sys.maxsize,
     ):
+        """Saves a checkpoint file for training
+        Arguments:
+            checkpoint_path {Union[str, pathlib.Path]} -- path to checkpoint directory, will be created if not present
+            checkpoint_freq {int} -- Number of epochs for checkpoint saving
+            prefix {str} -- Name of checkpoint file, will be {prefix}{current_epoch}.tar
+            start_epoch {int} -- For resuming training you can either use Trainer.load_from_latest or specify the epoch
+             here
+            max_chk {int} -- Max number of concurrent checkpoint files in the directory.
+        """
         self.checkpoint_path = pathlib.Path(checkpoint_path)
         self.checkpoint_freq = checkpoint_freq
         self.prefix = prefix
-        self.epoch_counter = 0
+        self.epoch_counter = start_epoch
```

Comment:
This is overwritten at each `on_epoch_end` call and I don't think we need it as a class field. I would use (as somehow you did) the epoch parameter passed to the method at each call given that `Trainer` handle the right epoch number both when training from scratch and when resuming with a trained model (done in `load_from_latest`

Reply:
removed in 6e029ca